### DISCUSSION TITLE
Documentation Survey

### DISCUSSION THREAD

**Discussion by Montoya** [MetaMask/snaps#442] [General]
Hello Snaps devs! We are working on improving our documentation and would like some feedback from all of you. Please reply with your responses to the following 3 questions: 

1. Where did you first hear about or read about Snaps?
2. What are some examples of documentation that you think are very effective?
3. How would you improve our Snaps Introduction documentation? [https://docs.metamask.io/guide/snaps.html](https://docs.metamask.io/guide/snaps.html)

Thanks!

**Comment by shawnmitchell** [comment]
1) Twitter

2) I like to start from a working example, so for me the examples were very helpful

3) The one thing that was not clear to me in the beginning was the need to delete a Snap and re-install on code changes.

**Reply by Montoya** [reply]
Thank you. In the latest version of Flask you should not have to delete and re-install anymore. Let me know if you still see the issue.

**Comment by onetrickwolf** [comment]
1. Presentation on Snaps by MetaMask devs
2. Working examples for all methods are great. Mimics MM docs which I was already familiar which was nice. Documentation generally clear and formatted well (e.g. obvious where methods can be called, etc). Development guide is very useful.
3. Honestly pretty good right now, maybe some info on how to set up development with TypeScript or how to set up an efficient development flow (e.g. I don't think it's mentioned anywhere that the snaps cli has a watch command). An example of how to set up a DAPP outside of the immediate snap might be nice too (I just used an env file but I wasn't sure if there was a better way). I had some issues with the configs too like setting different ports when developing multiple snaps didn't seem to work how I expected maybe more info on that.

**Comment by piotr-roslaniec** [comment]
1. A colleague mentioned it to me
2. Examples, recipes, tutorials. I'd love to see how MM Snaps work in depth (iframes?) in the advanced section. Also, a guide on deploying snaps to production.
3. Always up to date instructions on how to build MM Flask (which version, source, etc.).

**Comment by andyv09** [comment]
1. A colleague mentioned it to me.
2. Code examples and tutorials.
3. Add more examples of use, and keep documentation up to date with the changes on GitHub. Maybe add a 'good practices' section and a guide on how to use it with TypeScript. Another thing missing is a guide on how to deploy Snap to npm for example. And maybe a section on how to use it with dApps. When starting our project, I spent a lot of time going through the Filecoin Snap repo, because some of the previously mentioned things were missing from documentation (Docs have improved, since then). A roadmap of upcoming features would be lovely.
